### PR TITLE
CI-394: Change start and end dates of the time period.

### PR DESCRIPTION
**Pull request by SerhiiBorozenets**
[CI-394](https://cloverpop-internship.atlassian.net/browse/CI-394)

[CI-394]: https://cloverpop-internship.atlassian.net/browse/CI-394?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by rogergraves — COMMENTED** [review]
Are these supposed to be hard-coded versus being set in ENV variables?

**Comment by SerhiiBorozenets** [comment]
> Are these supposed to be hard-coded versus being set in ENV variables?

@rogergraves, in our current implementation it was done like this.  If necessary, we can add the use of env variables.  Do you think it needs to be done?

**Comment by rogergraves** [comment]
> > Are these supposed to be hard-coded versus being set in ENV variables?
> 
> @rogergraves, in our current implementation it was done like this. If necessary, we can add the use of env variables. Do you think it needs to be done?

@SerhiiBorozenets if it's something that might be changed by the org then ENV variable is better since it can be done without a coding change. If you think it's unlikely to be changed then it's OK to keep it as a code setting.

File: spec/models/time_period_spec.rb [hunk 1]
```diff
@@ -51,14 +51,19 @@
   describe 'model methods' do
     before(:each) do
       TimePeriod.destroy_all
+      allow(ENV).to receive(:[]).with('START_WEEK_DAY').and_return('monday')
```
Comment (SBorozenets) [inline spec/models/time_period_spec.rb:54]: you also can use `stub_const('ENV', {'START_WEEK_DAY' => 'monday'})` but your version is good too

**Review by SBorozenets — APPROVED** [review]

File: app/models/time_period.rb [hunk 2]
```diff
@@ -53,7 +53,8 @@ def previous_time_period
     def create_time_period
       return if current_time_period.present?
 
-      start_date = Date.current.beginning_of_week(:sunday)
+      start_week = ENV['START_WEEK_DAY'].present? ? ENV['START_WEEK_DAY'].to_sym : :monday
```
Comment (rogergraves) [inline app/models/time_period.rb:56]: Suggestion: 
```
ENV['START_WEEK_DAY']&.to_sym || :monday
```

**Review by rogergraves — APPROVED** [review]

**Pull request merged**